File: micronaut-hello-world/src/main/java/hello/world/HelloController.java
package hello.world;

import io.micronaut.http.annotation.Controller;
import io.micronaut.http.annotation.Get;
import io.micronaut.http.annotation.Post;

import javax.validation.Valid;

import static io.micronaut.http.MediaType.TEXT_PLAIN;

@Controller("/hello")
public class HelloController {
    /**
     * Announces ourselves.
     *
     * @return A greeting
     */
    @Get(produces = TEXT_PLAIN)
    public String index() {
        return "Hello World";
    }

    /**
     * Makes a roundtrip.
     *
     * @param roundTripData Garbage in
     *
     * @return Garbage out
     */
    @Post
    public RoundTripData roundTrip(final @Valid RoundTripData roundTripData) {
        return roundTripData;
    }
}

File: kotlin-micronaut/src/main/kotlin/x/micronaut/HelloController.kt
package x.micronaut

import io.micronaut.http.MediaType.TEXT_PLAIN
import io.micronaut.http.annotation.Controller
import io.micronaut.http.annotation.Get

@Controller("/hello")
class HelloController {
    @Get(produces = [TEXT_PLAIN])
    fun greeting(): String {
        return "Hello World"
    }
}
